### DISCUSSION TITLE
How do I activate hostap(hotspot) in my code?

### DISCUSSION THREAD

**Discussion by wooseokyourself** [RaspAP/raspap-webgui#756] [Q&A] [answered]
Hi. I wanna use this app in my python project.

All I want to do is this; normally the hotspot is dead, and when some condition my program should activates the hotspot of my raspi.

So I had found some API for developers but I couldn't find them. 

Maybe it seems impossible cuz I have no experience for backend of web service.

Can anyone help me?

**Comment by billz** [comment, marked as answer]
The shell script [installers/servicestart.sh](https://github.com/billz/raspap-webgui/blob/master/installers/servicestart.sh) is executed when the hotspot is started in the UI. You can execute it with Python by doing something like this:

```
import os
returncode = os.system('sudo /etc/raspap/hostapd/servicestart.sh')
print("The exit code was: %d" % returncode)
```

**Reply by wooseokyourself** [reply]
Hi! Thanks for your answer and it works!!

However, is there anyway to stop? Just ```sudo systemctl stop hosted.service``` is fine?

**Reply by wooseokyourself** [reply]
I just checked all I wanted to do is ```sudo system ctl stop hostapd.service``` and ```sudo system ctl start hostapd.service```. Thank you for sharing such a great project!

**Reply by billz** [reply]
Yes you can start/stop hostapd directly like this too. The RaspAP servicestart script manages 3 related services (hostapd, dnsmasq & dhcpcd) and brings them up in a specific order and timing. You can use the `--action stop` parameter with this script to shut them down.

**Reply by wooseokyourself** [reply]
Nice answer and I got it. Thanks!
